src/backend/base/langflow/main.py

```
@@ -152,10 +152,25 @@
             current_time = asyncio.get_event_loop().time()
             logger.debug("Creating/updating starter projects")
-            await create_or_update_starter_projects(all_types_dict)
-            logger.debug(f"Starter projects updated in {asyncio.get_event_loop().time() - current_time:.2f}s")
+            import tempfile
+            from filelock import FileLock
+            lock_file = Path(tempfile.gettempdir()) / "langflow_starter_projects.lock"
+            lock = FileLock(lock_file, timeout=1)
+            try:
+                with lock:
+                    await create_or_update_starter_projects(all_types_dict)
+                    logger.debug(f"Starter projects created/updated in {asyncio.get_event_loop().time() - current_time:.2f}s")
```

Ruff (0.11.9)

Line too long (126 > 120)

(E501)